File: api/internal/domain/favorite_group.go
package domain

import "time"

// FavoriteGroup represents a user's bookmark of a group
type FavoriteGroup struct {
	ID        string    `json:"id"`
	DeviceID  string    `json:"device_id"`
	GroupID   string    `json:"group_id"`
	CreatedAt time.Time `json:"created_at"`
}

File: web/src/shared/domain/favorite_group.ts
/**
 * Favorite Group domain model
 * Represents a user's bookmark of a group
 */

export interface FavoriteGroup {
  id: string; // NanoID (21 chars)
  device_id: string; // NanoID (21 chars)
  group_id: string; // NanoID (21 chars)
  created_at: string; // ISO timestamp
}

export interface FavoriteGroupCreateRequest {
  group_id: string; // NanoID (21 chars)
}

